web/src/popover_menus.js

```diff
@@ -307,13 +307,35 @@ export function initialize() {
                 return;
             }
 
-            $popper.one("click", ".sidebar-popover-mute-topic", () => {
-                muted_topics_ui.mute_topic(stream_id, topic_name);
+            $popper.one("click", ".sidebar-popover-toggle-unmute", () => {
+                const new_visibility_policy = user_topics.topic_has_visibility_policy(
+                    stream_id,
+                    topic_name,
+                    user_topics.visibility_policy.UNMUTED,
+                )
+                    ? user_topics.visibility_policy.INHERIT
+                    : user_topics.visibility_policy.UNMUTED;
+                user_topics.set_user_topic_visibility_policy(
+                    stream_id,
+                    topic_name,
+                    new_visibility_policy,
+                );
                 instance.hide();
             });
 
-            $popper.one("click", ".sidebar-popover-unmute-topic", () => {
-                muted_topics_ui.unmute_topic(stream_id, topic_name);
+            $popper.one("click", ".sidebar-popover-toggle-mute", () => {
```

Comment:
I think it'd be better to use a different CSS class name and click handler for the "unmute" and "mute" cases; squashing the toggling behavior together feels hard to read.